src/cinder/Signals.cpp

```diff
@@ -111,6 +111,22 @@ ScopedConnection& ScopedConnection::operator=( ScopedConnection &&rhs )
 	return *this;
 }
 
+ConnectionList::~ConnectionList()
+{
+	clear();
+}
+
+void ConnectionList::add( Connection &&target )
+{
+	mConnections.emplace_back( target );
+}
+
+void ConnectionList::clear()
+{
+	for( auto &conn : mConnections )
```

Comment:
@MikeGitb great advice, that makes sense to me so I'm in the middle of implementing it this way. One question comes up though, does this seem appropriate?

```cpp
Connection& Connection::operator=( Connection &&rhs )
{	
	mDisconnector = move( rhs.mDisconnector );
	rhs.mDisconnector.reset(); // <-- why is this needed?

	mLink = move( rhs.mLink );
	rhs.mLink = nullptr;

	mPriority = move( rhs.mPriority );
	rhs.mPriority = 0;

	return *this;
}
```
This seems wrong to me according to the weak_ptr [docs](http://en.cppreference.com/w/cpp/memory/weak_ptr/weak_ptr), which state that once I've moved a weak_ptr then it will be empty. However If I don't explicitly `.reset()` mDisconnector there, then it still has a valid weak reference, and its `use_count()` reports 1 (in my unit test).

Here's an [SO post](http://stackoverflow.com/a/26581337/506584) explaining the lack of implicit generation of move constructors and move assignment operators in VS2013. I think we have to make this work for that compiler, so I'm fine with adding in the extra constructors for now, just want to make sure to get them right.